code/game/objects/structures/roguetown/mobspawner.dm

```diff
@@ -31,7 +32,9 @@ var/global/max_total_spawned_mobs = 100 // New global variable for the total lim
 		if (total_spawned_mobs < max_total_spawned_mobs && current_spawned_mobs < max_spawned_mobs)
 			spawn_random_mobs(mobs_to_spawn)
 		start_spawning()
-
+		if (temporary == TRUE) 
+			sleep(30)
+			del src
```

Comment:
NO NO NO NO NO NO NO NO
NEVER DO THIS
NEVER EVER EVER
NEVER EVER USE DEL DIRECTLY, ONLY EVER QDEL
```suggestion
		if (temporary) 
			QDEL_IN(3 SECONDS)
			return
```